seaweedfs/src/main/java/site/ycsb/db/SeaweedClient.java

```diff
@@ -0,0 +1,301 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ * <p>
+ * SeaweedFS storage client binding for YCSB.
+ */
+package site.ycsb.db;
+
+import seaweedfs.client.*;
+import site.ycsb.*;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+import org.codehaus.jackson.JsonFactory;
+import org.codehaus.jackson.JsonGenerator;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.ObjectNode;
+
+/**
+ * SeaweedFS Storage client for YCSB framework.
+ *
+ * The size of the file to upload is determined by two parameters:
+ * - fieldcount this is the number of fields of a record in YCSB
+ * - fieldlength this is the size in bytes of a single field in the record
+ * together these two parameters define the size of the file to upload,
+ * the size in bytes is given by the fieldlength multiplied by the fieldcount.
+ * The name of the file is determined by the parameter key.
+ * This key is automatically generated by YCSB.
+ */
+public class SeaweedClient extends DB {
+
+  protected static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private FilerClient filerClient;
+  private FilerGrpcClient filerGrpcClient;
+  private String filerHost;
+  private int filerPort;
+  private String folder;
+
+  /**
+   * Cleanup any state for this storage.
+   * Called once per instance;
+   */
+  @Override
+  public void cleanup() throws DBException {
+  }
+
+  /**
+   * Delete a file from SeaweedFS Storage.
+   *
+   * @param tableName The name of the table
+   * @param key  The record key of the file to delete.
+   * @return OK on success, otherwise ERROR. See the
+   * {@link DB} class's description for a discussion of error codes.
+   */
+  @Override
+  public Status delete(String tableName, String key) {
+    if (!filerClient.rm(this.folder + "/" + tableName + "/" + key, true, true)) {
+      return Status.ERROR;
+    }
+    return Status.OK;
+  }
+
+  /**
+   * Initialize any state for the storage.
+   * Called once per SeaweedFS instance; If the client is not null it is re-used.
+   */
+  @Override
+  public void init() throws DBException {
+    filerHost = getProperties().getProperty("seaweed.filerHost", "localhost");
+    filerPort = Integer.parseInt(getProperties().getProperty("seaweed.filerPort", "8888"));
+    folder = getProperties().getProperty("seaweed.folder", "/ycsb");
+    filerGrpcClient = new FilerGrpcClient(filerHost, filerPort+10000);
+    filerClient = new FilerClient(filerGrpcClient);
+    filerClient.mkdirs(this.folder, 0755);
```

Comment:
this is going to happen once per YCSB client thread. just want to make sure that's okay?

Reply:
this is not ideal but ok. Could not find a better place to setup test initialization..